src/clarity-angular/datagrid/datagrid.ts

```diff
@@ -223,9 +227,18 @@ export class Datagrid implements AfterContentInit, AfterViewInit, OnDestroy {
     ngAfterViewInit() {
         // TODO: determine if we can get rid of provider wiring in view init so that subscriptions can be done earlier
         this.triggerRefresh();
-        this._subscriptions.push(this.sort.change.subscribe(() => this.triggerRefresh()));
-        this._subscriptions.push(this.filters.change.subscribe(() => this.triggerRefresh()));
-        this._subscriptions.push(this.page.change.subscribe(() => this.triggerRefresh()));
+
+        // these changes (sort, filters, page) all should call triggerRefresh. The triggerRefresh
+        // function takes a snapshot of the most recent state and emits the refresh event. Here, we combine
+        // and debounce them so that the rapid successive changes don't throw the infamous
+        // "expression has changed after it was checked" error.
+        let merged = Observable.merge(this.sort.change, this.filters.change, this.page.change)
+            .debounce(() => Observable.interval(300));
```

Comment:
The two states that produce the EHCAIWC are different (but I think the second state change fires mid-cycle of the first refresh?), so not re-emitting duplicate states won't fix this problem. I'm willing to dig this a bit more to see if there are alternative solutions rather than debounce.